source/routing/isochron.cpp

```diff
@@ -0,0 +1,147 @@
+    if (direction < 0) {
+        direction_rad = 2 * M_PI - direction_rad;
+    }
+    double center_lat_deg = center.lat();
+    double center_lon_deg = center.lon();
+    type::GeographicalCoord center_right_interval = type::GeographicalCoord(center_lon_deg, center_lat_deg);
+    double center_lat_rad = center_right_interval.lat() * GeographicalCoord::N_DEG_TO_RAD;
+    double center_lon_rad = center_right_interval.lon() * GeographicalCoord::N_DEG_TO_RAD;
+    double lat;
+    double lon;
+    // To avoid rounding error we use ° instread of rad.
+    if (fabs(center_right_interval.lat()) == 90) {
+        lon = direction_rad - M_PI;
+        lat = M_PI_2 - alpha;
+    } else {
+        double delta_lat;
+        double delta_lon;
+        if (fmod(direction, 180) == 0) {
+            delta_lat = pow(-1, fmod(direction / 180, 2)) * alpha;
+            lat = center_lat_rad + delta_lat;
+            lon = center_lon_rad;
+        } else {
+            // asin and acos give results in rad.
+            double projection = asin(sin(alpha) * sin(direction_rad));
+            delta_lat = acos(cos(alpha) / cos(projection));
+            if (direction_rad > M_PI_2 && direction_rad < 3 * M_PI_2) {
+                delta_lat = -1 * delta_lat;
+            }
+            lat = center_lat_rad + delta_lat;
+            if (lat == M_PI_2) {
+                delta_lon = (direction == 90 ? 1 : - 1) * alpha;
+            } else {
+                double b = (cos(alpha) - sin(lat) * sin(center_lat_rad)) / (cos(lat) * cos(center_lat_rad));
+                if (fabs(b) >= 1) {
+                    b = (std::signbit(b) ? fmod(b - 1, 2) + 1 : fmod(b + 1, 2) - 1);
+                }
+                delta_lon = acos(b);
+                if (direction_rad > M_PI) {
+                    delta_lon = -1 * delta_lon;
+                }
+            }
+            lon = center_lon_rad + delta_lon;
+        }
+    }
+    // Lon and lat are in °
+    double lon_deg = lon * N_RAD_TO_DEG;
+    double lat_deg = lat * N_RAD_TO_DEG;
+    return type::GeographicalCoord(lon_deg, lat_deg);
+}
+
+
+type::Polygon circle(const type::GeographicalCoord& center,
+                     const double& radius) {
+    type::Polygon points;
+    auto& points_out = points.outer();
+    points_out.reserve(360);
+    for (double i = 0; i < 360; i++) {
+        points_out.push_back(project_in_direction(center, i, radius));
+    }
+    points_out.push_back(project_in_direction(center, 0, radius));
+    return points;
+}
+
+type::MultiPolygon build_isochron(RAPTOR& raptor,
+                                const std::vector<type::StopPoint*>& stop_points,
+                                const bool clockwise,
+                                const DateTime& init_dt,
+                                const DateTime& bound,
+                                const map_stop_point_duration& origin) {
+    type::MultiPolygon circles;
+    double speed = 0.8; // About 3km/h
```

Comment:
not a fan of that hard-coded value, but I guess you're gonna change it in another PR.
Maybe add a todo?
Or even already add a param for that?